examples/general/abs_home_path/src/lib.rs

```diff
@@ -178,52 +124,94 @@ fn ui() {
         .run();
 }
 
+// Combined test for context allowances (build script and test context)
 #[test]
-fn build_script_allowance() {
+fn context_allowance() {
     use dylint_internal::CommandExt;
     use std::{
         io::{Write, stderr},
         path::Path,
         process::Command,
+        process::Output,
     };
 
-    // Skip test if repository is not stored in the user's home directory
+    struct TestCase {
+        manifest_path: &'static str,
+        context_name: &'static str,
+        assert_fn: fn(&Output),
+    }
```

Comment:
I think `TestCase` needs an additional field to determine whether the following should apply:
https://github.com/trailofbits/dylint/blob/ae845edfa22b582bf5d568e622db3607080d706e/examples/general/abs_home_path/src/lib.rs#L208
Note that the original `build_script_allowance` test applied this.